packages/core/src/machines/index.ts

```diff
@@ -683,6 +714,22 @@ export const createAuthMachine = ({
             otp: data.otp
           })
         },
+        signInSecurityKeyEmail: async (_, { email }) => {
+          if (!isValidEmail(email)) {
+            throw new CodifiedError(INVALID_EMAIL_ERROR)
+          }
+          const options = await postRequest<PublicKeyCredentialRequestOptionsJSON>(
+            '/signin/webauthn',
+            { email }
+          )
+          let credential: AuthenticationCredentialJSON
+          try {
+            credential = await startAuthentication(options)
+          } catch (e) {
+            throw new CodifiedError(e as Error)
+          }
+          return postRequest<SignInResponse>('/signin/webauthn/verify', { email, credential })
+        },
```

Comment:
What's the flow if users must verify their email? Is it the following:

- Fill in email in sign up form
- Click submit
- Get prompted to add their security key (e.g. with face id)
- Email is sent to the user
- User clicks on the link in the email to verify their email
- User is signed in?

Reply:
- fill in email
- click submit
- email is not verified: does not go further
- up to the frontend to decide what to do

Reply:
Wouldn't it be better if the user could add their security keys but not sign in until the email is verified?

Just like email + password. The user signs up using email and password and then needs to verify their email to sign in.

Reply:
Not sure I understood, but the two ways I did, no, it won't be better:
- if by "add security key" you mean try to authenticate with the key, if the email is not verified, but connected still the user would not have been able to add their key to their account - no key is available.
- if you mean "add security key", we don't add it on sign up, and we shouldn't security-wisde: while the email is not verified yet, a malicious person could then add a security key while waiting for the real user to validate their email. Once done, the malicious person would become able to authenticate

Reply:
Sorry, let me rephrase it.

I'm thinking we should treat a security key like a password.

A user signs up with email+password, or email+security key. 

If the user must verify their email before signing in, they do so.

Once the email verification step is completed, the user can sign in with email+password or email+security key.

Reply:
sign in:
the frontend can decide to either send an alert like "you must verify your email first" or send a new verification email with `nhost.auth.sendVerificationEmail`.
It follows the same patterns as a user trying to authenticate with an unverified email and a password

sign up:
- create an user with email+password, passwordless email, oauth - whatever method that gets an email
- authenticate
- add a sign-in key

We talked about removing the webauthn sign-up endpoint two weeks ago and it was basically the main refactoring of auth v0.11, as the logic was a duplicate to sign-up with email+password and wouldn't allow other registration methods

Reply:
We agreed offline to keep things as it is. 

- Users can add a security key once they are signed in.
- When users have added a security key, they can sign in using their email and security key.

Later, we might come back to allow users to sign up with an email + security key just like they would using an email + password.